sql/information_schema/columns_table.go

```diff
@@ -125,54 +125,106 @@ func columnsRowIter(ctx *sql.Context, cat sql.Catalog, columnNameToDefault map[s
+					if st, ok := c.Type.(sql.StringType); ok {
+						charMaxLen = st.MaxCharacterLength()
+					}
+					dataType = strings.TrimSuffix(dataType, fmt.Sprintf("(%v)", charMaxLen))
 				}
 				ordinalPos = uint64(i + 1)
 
 				fullColumnName := db.Name() + "." + t.Name() + "." + c.Name
 				colDefault = trimColumnDefaultOutput(columnNameToDefault[fullColumnName])
+				colType = strings.ToLower(c.Type.String())
+				if c.Type == sql.Boolean {
+					colType = colType + "(1)"
+				}
 				columnKey := ""
+				// PK is checked here first because there are PKs from table implementations that don't implement sql.IndexedTable
 				if c.PrimaryKey {
 					columnKey = "PRI"
+				} else if val, ok := columnKeyMap[c.Name]; ok {
+					columnKey = val
+					// A UNIQUE index may be displayed as PRI if it cannot contain NULL values and there is no PRIMARY KEY in the table
+					if !c.Nullable && !hasPK && columnKey == "UNI" {
+						columnKey = "PRI"
+					}
 				}
 
 				rows = append(rows, sql.Row{
-					"def",                            // table_catalog
-					db.Name(),                        // table_schema
-					t.Name(),                         // table_name
-					c.Name,                           // column_name
-					ordinalPos,                       // ordinal_position
-					colDefault,                       // column_default
-					nullable,                         // is_nullable
-					strings.ToLower(c.Type.String()), // data_type
-					nil,                              // character_maximum_length
-					nil,                              // character_octet_length
-					nil,                              // numeric_precision
-					nil,                              // numeric_scale
-					nil,                              // datetime_precision
-					charName,                         // character_set_name
-					collName,                         // collation_name
-					strings.ToLower(c.Type.String()), // column_type
-					columnKey,                        // column_key
-					c.Extra,                          // extra
-					"select",                         // privileges
-					c.Comment,                        // column_comment
-					"",                               // generation_expression
+					"def",      // table_catalog
+					db.Name(),  // table_schema
+					t.Name(),   // table_name
+					c.Name,     // column_name
+					ordinalPos, // ordinal_position
+					colDefault, // column_default
+					nullable,   // is_nullable
+					dataType,   // data_type
+					charMaxLen, // character_maximum_length
+					nil,        // character_octet_length
+					nil,        // numeric_precision
+					nil,        // numeric_scale
+					nil,        // datetime_precision
+					charName,   // character_set_name
+					collName,   // collation_name
+					colType,    // column_type
+					columnKey,  // column_key
+					c.Extra,    // extra
+					"select",   // privileges
+					c.Comment,  // column_comment
+					"",         // generation_expression
```

Comment:
Awesome that we have an easy way to see what fields are left to be populated in information_schema.columns!

Could you add `SRS_ID` as another unpopulated field while you're in this file? I noticed that column is part of this table, and it'd be nice to go ahead and add it in so we can see the full set of columns we're supposed to support for MySQL compliance. You don't need to populate that field yet of course, but it would be nice to start tracking it.